docs/src/implemented-proposals/tower-bft.md

```diff
@@ -67,21 +86,26 @@ At slot 10 the new votes caught up to the previous votes. But _vote 2_ expires a
 |    7 |        11 |       2 |                   13 |
 |    1 |         1 |      16 |                   17 |
 
-The lockout for vote 1 will not increase from 16 until the stack contains 5 votes.
-
-### Slashing and Rewards
-
-Validators should be rewarded for selecting the fork that the rest of the cluster selected as often as possible. This is well-aligned with generating a reward when the vote stack is full and the oldest vote needs to be dequeued. Thus a reward should be generated for each successful dequeue.
+The lockout for vote 1 will not increase from 16 until the tower contains 5 votes.
 
 ### Cost of Rollback
 
 Cost of rollback of _fork A_ is defined as the cost in terms of lockout time to the validator to confirm any other fork that does not include _fork A_ as an ancestor.
 
 The **Economic Finality** of _fork A_ can be calculated as the loss of all the rewards from rollback of _fork A_ and its descendants, plus the opportunity cost of reward due to the exponentially growing lockout of the votes that have confirmed _fork A_.
 
-### Thresholds
+### Threshold Check
+In order to prevent a validator from locking themselves out on the wrong fork
+in the case of a partition, there also needs to be a check to ensure that the rest of the cluster is committing to the same fork. This check is called the
+"threshold check", and is outlined as follows.
+
+In deciding whether to vote for a block `B`:
 
-Each validator can independently set a threshold of cluster commitment to a fork before that validator commits to a fork. For example, at vote stack index 7, the lockout is 256 slots. A validator may withhold votes and let votes 0-7 expire unless the vote at index 7 has at greater than 50% commitment in the cluster. This allows each validator to independently control how much risk to commit to a fork. Committing to forks at a higher frequency would allow the validator to earn more rewards.
+1. Simulate a vote for `B` on your current tower
+2. Pop off all the votes that would be expired by `B`
+3. Now index every vote in the tower from `[0, tower.length()]`, assuming that the most recent simulated vote `B` is index 0, the second most recent vote is index 1, etc.
+4. Let `threshold_check_depth = 8` (the "depth" of 8 refers to this), now let `T` be the vote in the tower with index equal to threshold_check_depth.
```

Comment:
Is thredshold_check_depth always 8?
Then you can change it to:
Let `T` be ... threshold_check_depth (currently hardcoded to be 8).